bin/canary_delete_an_incident.py

```diff
@@ -20,7 +22,10 @@ def validate_params(self):
             self.log_error('canary_domain is a mandatory setup parameter, but its value is None.')
             return False
 
-        if not self.get_global_setting("api_key"):
+        splunk_session_key = self.session_key
+        splunkService = client.connect(token=splunk_session_key)
+        api_key = get_api_key(splunkService)
+        if not api_key:
             self.log_error('api_key is a mandatory setup parameter, but its value is None.')
             return False
```

Comment:
What is the difference between these two patterns?

```
from api_key_retrieval import get_api_key
api_key = get_api_key(helper.service)
```

```
splunk_session_key = self.session_key
splunkService = client.connect(token=splunk_session_key)
api_key = get_api_key(splunkService)
```

Reply:
In `TA-canary/bin/input_module_canary_api.py` and `TA-canary/bin/input_module_canary_daily_poll.py` the `helper` object is of a different type to the `helper` object in the following 4 files:
```
TA-canary/bin/ta_canary/modalert_canary_acknowledge_incident_helper.py
TA-canary/bin/ta_canary/modalert_canary_delete_an_incident_helper.py
TA-canary/bin/canary_acknowledge_incident.py
TA-canary/bin/canary_delete_an_incident.py
```

In the latter 4 files, we have to use the session_key value that comes with the helper object to create a splunk service type object. That splunk service type object is the one that can then call the storage_passwords function to get the decrypted API key.